api/src/main/java/bio/terra/tanagra/service/search/Query.java

```diff
@@ -0,0 +1,27 @@
+package bio.terra.tanagra.service.search;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Optional;
+
+/** A search query on an {@link Entity}. */
+@AutoValue
+public abstract class Query {
+  /** What to select for the query. */
+  // TODO consider selections with a different entity than the primary entity.
+  public abstract ImmutableList<Selection> selections();
+
+  /** The primary entity being queried. */
+  public abstract Entity primaryEntity();
+
+  /** The filter to apply to the primary entity, if there is a filter. */
+  public abstract Optional<Filter> filter();
```

Comment:
Looks like this is probably covered by the above TODO, but most of what the dataset builder (specifically via concept sets) is doing currently is applying filters to related entities. But first joined onto an initial cohort definition, i.e. set of person rows defined by a filter.

Not sure how this is modeled here.

Reply:
Yep, very much TODO'd. right now I'm expecting it to be a subtype of filter.